Engine/Settings.cs

```diff
@@ -497,6 +497,13 @@ private static bool IsBuiltinSettingPreset(object settingPreset)
         internal static SettingsMode FindSettingsMode(object settings, string path, out object settingsFound)
         {
             var settingsMode = SettingsMode.None;
+
+            // if the provided settings argument is wrapped in an expressions then PowerShell resolves it but it will be of type PSObject and we have to operate then on the BaseObject
+            if (settings is PSObject)
+            {
+                settings = ((PSObject)settings).BaseObject;
+            }
```

Comment:
```suggestion
            if (settings is PSObject settingsFoundPSObject)
            {
                settings = settingsFoundPSObject.BaseObject;
            }
```

I believe you can do this rather than cast it all over again.